ROADMAP.md

```diff
@@ -0,0 +1,135 @@
+# ROADMAP
+
+See [README.md](README.md) for developer getting start instructions.
+
+## Minimal complete pipeline
+
+* `heareval/baseline.py` - Simple unlearned embedding algorithm
+following the API.
+
+* `heareval/tasks/` - One time preprocessing of tasks into train +
+test WAVs and `train/test.csv`.
+
+* `heareval/predict/` - [not implemented] For each task, map from
+API-compatible embeddings to `predicted-test.csv`. (This involves
+a linear model for most task types.)
+
+* `heareval/evaluate/` - Given task type, `test.csv`, and
+`predicted-test.csv` output evaluation scores.
+(Note that we can implement this now just by creating random `test.csv`
+and `predicted-test.csv` files, Christian is starting this task.)
+
+## Task Types + Prediction
+
+See [heareval/tasks/README.md](heareval/tasks/README.md) for more
+details.
+
+Ideally, `predicted-test.csv` would have identical format as
+`test.csv`. In practice, for AUC which depends upon the predicted
+score, we might need to change the format slightly.
+
+### Learning Based Tasks
+
+Entire audio, multiclass:
+    * Concat all embeddings for the audio, make a multiclass prediction.
+
+Entire audio, multilabel:
+    * Concat all embeddings for the audio, make a multilabel prediction.
+
+Frame-based multilabel (transcription + sound event detection):
+    * For each frame, make a multilabel prediction.
+
+### Unlearned tasks
+
+JND (audio1 vs audio2 has a perceptible difference):
+    * np.abs(embedding1 - embedding2), where embedding1 is the
+    concatenation of all audio1 frames, same for embedding2.
+    * Compute AUC
+
+Ranking tasks:
+    * This requires more thought. There are two options:
+	    1) Instead of each train and test row being a list of
+	    files in ranked order, we could convert it to a long
+	    set of file triples where the ordering is known.
+    	This is a bit uglier and less general but simpler to work
+    	with.
+        2) We could perform a forced ranking on all the files.
+        This makes the prediction code for ranking more gnarly,
+        but the `predict-train.csv` is clean and consistent and
+        we can just run Spearman correlation for eval.
+    * Regardless of the final output of the predictor, we will
+    probably need triplets.
```

Comment:
I've mentioned this before, but I like the idea of a simple list of distances from each sound to one anchor file.

Reply:
Which anchor file? The middle one? Should we use prior knowledge and cheat a bit when doing the ranking?

We could also do the ranking versus ALL the files, and average the ranks.

Reply:
The "lowest" or "highest" incidence file, I guess? Assume one-dimension of variation (which I guess is implicit in a ranking task anyway, no?)